exchanges/gateio/gateio_wrapper.go

```diff
@@ -495,21 +490,11 @@ func (g *Gateio) GetDepositAddress(cryptocurrency currency.Code, _ string) (stri
 		return "", err
 	}
 
-	// Waits for new generated address if not created yet, its variable per
-	// currency
 	if addr == gateioGenerateAddress {
-		time.Sleep(10 * time.Second)
-		addr, err = g.GetCryptoDepositAddress(cryptocurrency.String())
-		if err != nil {
-			return "", err
-		}
-		if addr == gateioGenerateAddress {
-			return "", errors.New("new deposit address is being generated, please retry again shortly")
-		}
-		return addr, nil
+		return "",
+			errors.New("new deposit address is being generated, please retry again shortly")
```

Comment:
Nice catch, this should be handled by the calling application rather than holding up execution for 10 seconds